core/src/main/java/greencity/webcontroller/ManagementPlacesController.java

```diff
@@ -33,6 +49,59 @@ public String getAllPlaces(@RequestParam(required = false, name = "query") Strin
         PageableDto<AdminPlaceDto> allPlaces =
             query == null || query.isEmpty() ? placeService.findAll(pageable) : placeService.searchBy(pageable, query);
         model.addAttribute("pageable", allPlaces);
+        model.addAttribute("categoryList", categoryService.findAllCategoryDto());
+        List<String> discountSpecifications = specificationService.findAllSpecificationDto().stream().map(
+            SpecificationNameDto::getName).collect(Collectors.toList());
+        model.addAttribute("discountSpecifications", discountSpecifications);
         return "core/management_places";
     }
+
+    /**
+     * Method which saves {@link Place}.
+     *
+     * @param placeAddDto dto with info for registering place.
+     * @param user        is an admin
+     * @return {@link GenericResponseDto}
+     */
+    @PostMapping("/")
+    @ResponseBody
+    public GenericResponseDto savePlace(@Valid @RequestBody PlaceAddDto placeAddDto, BindingResult bindingResult,
+                                        @ApiIgnore @CurrentUser User user) {
+        if (bindingResult.hasErrors()) {
+            GenericResponseDto genericResponseDto = new GenericResponseDto();
```

Comment:
I have seen such code couple of times. It might be time to create a builder inside `GenericResponseDto` that accepts `BindingResult`.